### PR TITLE
feat(issue-stream): Update priority badge/selector

### LINKED ISSUE #77819: Issue Stream Table: Modify priority dropdown to use signal icons
This will de-emphasize priority a bit - the red color was very distracting.

Estimate: 0.5 days

Image

### LINKED ISSUE #78683: Issue Stream Table: Update priority selector design
Change the priority selector design to drop background colors and add new icon design

Estimate: 0.5 days

### PR DESCRIPTION
**Pull request by MichaelSun48**
closes #77819

Creates the new design for the issue priority badge under the `variant` prop within the `<GroupPriorityBadge/>` component. This design is currently under a feature flag, so it is not visible. 

Demo with flag manually set to true: 

https://github.com/user-attachments/assets/d6f388a7-7c21-498c-afd2-7fc2f6faef23

### CODE REVIEW

File: static/app/components/badge/groupPriority.tsx [hunk 1]
```diff
@@ -85,21 +88,39 @@ function useLastEditedBy({
...
   return PRIORITY_OPTIONS.map(priority => ({
     textValue: PRIORITY_KEY_TO_LABEL[priority],
     key: priority,
-    label: <GroupPriorityBadge priority={priority} />,
+    label: (
+      <GroupPriorityBadge
+        variant={hasStreamlinedIssueStream ? 'streamlined' : 'default'}
+        priority={priority}
+      />
+    ),
```
Comment (MichaelSun48) [inline static/app/components/badge/groupPriority.tsx:104]: Note that I have decided to keep the labels for the badges that are shown in the dropdown, unlike for the badges in the issue stream.

File: static/app/icons/iconCellSignal.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import {forwardRef} from 'react';
+import {useTheme} from '@emotion/react';
+
+import {SvgIcon, type SVGIconProps} from 'sentry/icons/svgIcon';
+
+interface Props extends SVGIconProps {
+  bars?: 0 | 1 | 2 | 3;
+}
+const IconCellSignal = forwardRef<SVGSVGElement, Props>(({bars = 3, ...props}, ref) => {
```
Comment (MichaelSun48) [inline static/app/icons/iconCellSignal.tsx:9]: Not attached to this name, feel free to give suggestions.

File: static/app/components/badge/groupPriority.tsx [hunk 3]
```diff
@@ -34,6 +35,8 @@ type GroupPriorityDropdownProps = {
 type GroupPriorityBadgeProps = {
   priority: PriorityLevel;
   children?: React.ReactNode;
+  showLabel?: boolean;
+  variant?: 'default' | 'streamlined';
```
Comment (MichaelSun48) [inline static/app/components/badge/groupPriority.tsx:39]: copying the "streamlined" convention from issue details, open to better names to differentiate the two

Comment (scttcper) [reply static/app/components/badge/groupPriority.tsx:39]: i'd vote for another name

File: static/app/icons/iconCellSignal.tsx (hunk 2 above)
Comment (malwilley) [reply static/app/icons/iconCellSignal.tsx:9]: Looks like material icons [has a similar name](https://fonts.google.com/icons?selected=Material+Symbols+Outlined:signal_cellular_alt:FILL@0;wght@400;GRAD@0;opsz@24&icon.query=signal&icon.size=24&icon.color=%235f6368), so seems okay to me.

Only other think I could think of would be IconSignalBars

File: static/app/components/badge/groupPriority.tsx [hunk 4]
```diff
@@ -85,21 +88,39 @@ function useLastEditedBy({
...
+    priority === PriorityLevel.HIGH ? 3 : priority === PriorityLevel.MEDIUM ? 2 : 1;
+
   return (
-    <StyledTag type={getTagTypeForPriority(priority)}>
-      {PRIORITY_KEY_TO_LABEL[priority] ?? t('Unknown')}
+    <StyledTag
+      type={variant === 'signal' ? 'default' : getTagTypeForPriority(priority)}
+      icon={variant === 'signal' && <IconCellSignal bars={bars} />}
+    >
+      {(showLabel && PRIORITY_KEY_TO_LABEL[priority]) ?? t('Unknown')}
```
Comment (malwilley) [inline static/app/components/badge/groupPriority.tsx:123]: Can you use `VisuallyHidden` here so that the text still exists for screen readers and tests?

**Review by malwilley — APPROVED** [review]

**Pull request merged**